src/cursor/cur_hs.c

```diff
@@ -501,7 +550,7 @@ __curhs_search_near(WT_CURSOR *cursor, int *exactp)
              * be more than one record away the end of our target key/timestamp range. Keep
              * iterating backwards until we land on our key.
              */
-            while ((ret = file_cursor->prev(cursor)) == 0) {
+            while ((ret = __wt_hs_cursor_prev(session, file_cursor)) == 0) {
```

Comment:
input parameter should be hs_cursor right ?

Reply:
No, we are walking the underlying cursor here.

Reply:
Why caller has to bother about underlying cursor? shouldn't the abstration take care of that?

Reply:
We are in the implementation of abstraction. I don't quite understand what you mean.

Reply:
Just to be clear I am just fixing an existing bug and this not directly related to title of the ticket.